src/lib/Lib/ExecutionContext.cpp

```diff
@@ -5,30 +5,164 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 // Copyright (c) 2023 Vinnie Falco (vinnie.falco@gmail.com)
+// Copyright (c) 2023 Krystian Stasiowski (sdkrystian@gmail.com)
 //
 // Official repository: https://github.com/cppalliance/mrdox
 //
 
 #include "ExecutionContext.hpp"
+#include "lib/AST/Bitcode.hpp"
+#include "lib/Metadata/Reduce.hpp"
+#include <mrdox/Metadata.hpp>
 
 namespace clang {
 namespace mrdox {
 
+namespace {
+
+// A standalone function to call to merge a vector of infos into one.
+// This assumes that all infos in the vector are of the same type, and will fail
+// if they are different.
+// Dispatch function.
+mrdox::Expected<std::unique_ptr<Info>>
```

Comment:
why the `mrdox` namespace qualifier?

Reply:
`ExecutionContext.cpp` includes the bitcode serialization headers, and we use `llvm::Expected` in some places in them. I'm using a qualified-id more so for clarity than it actually being necessary